projects/knora-ui/src/lib/viewer/values/link-value/link-value.component.ts

```diff
@@ -0,0 +1,169 @@
+import {Component, Inject, Input, OnChanges, OnDestroy, OnInit, SimpleChanges} from '@angular/core';
+import {BaseValueComponent} from '../base-value.component';
+import {CreateLinkValue, ReadLinkValue, ReadResource, UpdateLinkValue, KnoraApiConnection} from '@knora/api';
+import {Subscription} from 'rxjs';
+import {AbstractControl, FormBuilder, FormControl, FormGroup, ValidatorFn} from '@angular/forms';
+import {KnoraApiConnectionToken} from '../../../core';
+@Component({
+  selector: 'kui-link-value',
+  templateUrl: './link-value.component.html',
+  styleUrls: ['./link-value.component.scss']
+})
+export class LinkValueComponent extends BaseValueComponent implements OnInit, OnChanges, OnDestroy {
+  @Input() displayValue?: ReadLinkValue;
+  @Input() parentResource: ReadResource;
+  @Input() propType: string;
+  resources: ReadResource[];
+  restrictToResourceClass: string;
+  valueFormControl: FormControl;
+  commentFormControl: FormControl;
+  form: FormGroup;
+
+  valueChangesSubscription: Subscription;
+  // label cannot contain logical operations of lucene index
+  customValidators = [];
+
+  constructor(@Inject(FormBuilder) private fb: FormBuilder, @Inject(KnoraApiConnectionToken) private knoraApiConnection: KnoraApiConnection) {
+    super();
+  }
+
+  /**
+   * Displays a selected resource using its label.
+   *
+   * @param resource the resource to be displayed (or no selection yet).
+   * @returns
+   */
+  displayResource(resource: ReadResource | null): string {
+
+    // null is the initial value (no selection yet)
+    if (resource !== null) {
+      return resource.label;
+    }
+  }
+
+  /**
+   * Search for resources whose labels contain the given search term, restricting to to the given properties object constraint.
+   * this is to be used for update and new linked resources
+   *
+   * @param label to be searched
+   */
+  searchByLabel(searchTerm: string): ReadResource[] {
+    const linkType = this.parentResource.getLinkPropertyIriFromLinkValuePropertyIri(this.propType);
+    if (typeof linkType === 'string') {
```

Comment:
Why is that needed?

Reply:
because getLinkPropertyFromLinkValueIRI returns string || Error.

Reply:
without this check, the compiler complains about setting Error as index.

Reply:
oops, that's a bug  in the lib, will fix it

Reply:
fixed in https://github.com/dasch-swiss/knora-api-js-lib/pull/168

Reply:
I fixed it, I think you can safely remove that check.

Reply:
I think you can remove the check for string.